src/runtime_src/driver/xclng/drm/xocl/subdev/mb_scheduler.c

```diff
@@ -1526,19 +1523,7 @@ exec_execute_copybo_cmd(struct exec_core *exec, struct xocl_cmd *xcmd)
 static void
 exec_notify_host(struct exec_core *exec)
 {
-	struct list_head *ptr;
-	struct client_ctx *entry;
-	struct xocl_dev *xdev = exec_get_xdev(exec);
-
 	SCHED_DEBUGF("-> %s(%d)\n", __func__, exec->uid);
-
-	/* now for each client update the trigger counter in the context */
-	mutex_lock(&xdev->ctx_list_lock);
-	list_for_each(ptr, &xdev->ctx_list) {
-		entry = list_entry(ptr, struct client_ctx, link);
-		atomic_inc(&entry->trigger);
-	}
-	mutex_unlock(&xdev->ctx_list_lock);
 	/* wake up all the clients */
```

Comment:
Should we really wake up all clients as are doing today and have always done in exec_notify_host?   Part of the reason for having the client trigger was maybe to know which specific client should be awoken, though that part of the logic was never used.

Reply:
why do we wake up all clients today? did we run into any issue if we just wake up the one who sent the cmd?

Reply:
No.  We just never implemented the necessary hooks to just wake up the right client.

Reply:
xocl_cmd has access to client so we can improve this by waking up only the client which submitted this request. Need to enhance exec_notify_host() to take a second argument, client pointer--
exec_notify_host(exec, xcmd->client);

However, let's limit the current cleanup to bug fixes only. The wakeup improvement can be done in a subsequent patch?

Reply:
Yes, I think the wake up improvement should be a separate PR and needs more thoughts.

Maybe we should just get fd from exec BO, and exec wait on that fd, instead of waiting on a global fd to xocl.